File: php/447-number-of-boomerangs.php
<?php
class Solution 
{
    /**
     * @param Integer[][] $points
     * @return Integer
     */
    public function numberOfBoomerangs($points) 
    {
        $result = 0;
        foreach ($points as $k => $v) {
            //map
            $map = [];
            foreach ($points as $kk => $vv) {
                if ($kk != $k) {
                    $map[$this->getDistance($v, $vv)]++;
                }
            }
            foreach ($map as $m) {
                $result += $m * ($m-1);
            }
        }
        return $result;
    }
    private function getDistance($i, $j) 
    {
        return pow($i[0]-$j[0], 2) + pow($i[1]-$j[1], 2);
    }
}

//test
$s = new Solution();
echo $s->numberOfBoomerangs([[0,0],[1,0],[2,0]]);

File: c++/447-number-of-boomerangs.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int result = 0;
        for (int i = 0; i < points.size(); i++) {
            unordered_map<int,int> record;
            for (int j = 0; j < points.size(); j++) {
                if (j != i) {
                    record[dist(points[i], points[j])]++;
                }
            }
            for (unordered_map<int,int>::iterator iter = record.begin(); iter != record.end(); iter++) {
                result += iter->second*(iter->second-1);
            }
        }
        return result;
    }
private:
    int dist(const pair<int,int> &pa, const pair<int,int> &pb) {
        return (pa.first - pb.first) * (pa.first - pb.first) + (pa.second - pb.second) * (pa.second - pb.second);
    }
};
